internal/adapters/consul/sync_test.go

```diff
@@ -143,3 +145,50 @@ func TestHTTPRouteDiscoveryChain(t *testing.T) {
 		})
 	}
 }
+
+func TestConsulSyncAdapter_Sync(t *testing.T) {
+	t.Parallel()
+
+	ctx, cancel := context.WithCancel(context.Background())
+	consulSrv, err := testutil.NewTestServerConfigT(t, func(c *testutil.TestServerConfig) {
+		c.Connect = map[string]interface{}{"enabled": true}
+	})
+	require.NoError(t, err)
+
+	t.Cleanup(func() {
+		cancel()
+		_ = consulSrv.Stop()
+	})
+
+	cfg := api.DefaultConfig()
+	cfg.Address = consulSrv.HTTPAddr
+	consul, err := api.NewClient(cfg)
+	require.NoError(t, err)
+
+	adapter := NewConsulSyncAdapter(testutil.Logger(t), consul)
+
+	gateway := core.ResolvedGateway{
+		ID: core.GatewayID{
+			Service: "name1",
+		},
+		Listeners: []core.ResolvedListener{{
+			TLS: core.TLSParams{
+				MinVersion: "TLSv1_2",
+			},
+		}},
+	}
+
+	err = adapter.Sync(ctx, gateway)
+	require.NoError(t, err)
+	// TODO: wait for sync to complete - how?
+	// consulSrv.WaitForServiceIntentions(t)
+
+	// FIXME: Config entry not found for "ingress-gateway" / "name1"
+	entry, _, err := consul.ConfigEntries().Get(api.IngressGateway, "name1", nil)
+	require.NoError(t, err)
+	ingress, ok := entry.(*api.IngressGatewayConfigEntry)
+	require.True(t, ok)
+	require.NotNil(t, ingress)
+	fmt.Printf("%#v\n", ingress)
+	require.Equal(t, "TLSv1_2", ingress.TLS.TLSMinVersion)
```

Comment:
Is there a better way to wait for the "sync" operation to complete than using require.Eventually here? I noticed that consul/api has some WaitFor functions available, not sure if anything like that would be useful here.